### PR TITLE
Begin Decoupling the basic routing test from the Global resolver.

### PR DESCRIPTION
**Pull request by patricksrobertson**
It is generally agreed upon that human beings lack the ambition to
rewrite 3800 lines of test code in one sitting, which mirrors the
inability to review 3800 lines of code. In order to make progress
on the basic routing tests, this PR does:
* Creates a new module using ApplicationTestCase for the first ten
basic tests.
* Adds the module to a temporary file name
* Uses more modern convienence methods for determining currentPath.
* Asserts based upon template text instead of selector length.

refs #15058

### CODE REVIEW

File: packages/ember/tests/routing/decoupled_basic_test.js [hunk 1]
```diff
@@ -0,0 +1,214 @@
...
+
+  getController(name) {
+    return this.applicationInstance.lookup(`controller:${name}`);
+  }
+
+  get currentPath() {
+    return this.getController('application').get('currentPath');
+  }
+
+  ['@test warn on URLs note included in the route set'](assert) {
```
Comment (alexander-alvarez) [inline packages/ember/tests/routing/decoupled_basic_test.js]: not

File: packages/ember/tests/routing/decoupled_basic_test.js [hunk 2]
```diff
@@ -0,0 +1,214 @@
...
+      }
+    }));
+
+    return this.visit('/').then(() => {
+      let text = this.$('#troll').text();
+      assert.equal(text, "Megatroll", 'the homepage template was rendered');
+    });
+  }
+
+  [`@test an Alternate template will pull
```
Comment (alexander-alvarez) [inline packages/ember/tests/routing/decoupled_basic_test.js]: An alternate

File: packages/ember/tests/routing/decoupled_basic_test.js [hunk 3]
```diff
@@ -0,0 +1,214 @@
...
+      let text = this.$('p').text();
+
+      assert.equal(text, 'YES I AM HOME',
+        'The homepage template was rendered'
+      );
+    });
+  }
+
+  [`@test Model passed via renderTemplate model
+    is set as controller's model`](assert) {
```
Comment (mixonic) [inline packages/ember/tests/routing/decoupled_basic_test.js]: Can this not be wrapped? newline in a test name seems odd.

File: packages/ember/tests/routing/decoupled_basic_test.js [hunk 4]
```diff
@@ -0,0 +1,214 @@
...
+      let text = this.$('p').text();
+
+      assert.equal(text, 'Comes from home.',
+        'the homepage template was rendered from data from the HomeController'
+      );
+    });
+  }
+
+  [`@test The Homepage with explicit template name
+    in renderTemplate and controller`](assert) {
```
Comment (mixonic) [inline packages/ember/tests/routing/decoupled_basic_test.js]: Same here.

File: packages/ember/tests/routing/decoupled_basic_test.js [hunk 5]
```diff
@@ -0,0 +1,214 @@
...
+      let text = this.$('p').text();
+
+      assert.equal(text, 'Comes from homepage',
+        'the homepage template was rendered'
+      );
+    });
+  }
+
+  [`@test The template will pull in an
+    alternate controller via key/value`](assert) {
```
Comment (mixonic) [inline packages/ember/tests/routing/decoupled_basic_test.js]: And here

File: packages/ember/tests/routing/decoupled_basic_test.js [hunk 6]
```diff
@@ -0,0 +1,214 @@
...
+      let text = this.$('p').text();
+
+      assert.equal(text, 'Comes from homepage',
+        'the homepage template was rendered'
+      );
+    });
+  }
+
+  [`@test An alternate template will pull in an
+    alternate controller instead of controllername`](assert) {
```
Comment (mixonic) [inline packages/ember/tests/routing/decoupled_basic_test.js]: And here.

File: packages/ember/tests/routing/decoupled_basic_test.js [hunk 7]
```diff
@@ -0,0 +1,214 @@
...
+    }));
+
+    return this.visit('/').then(() => {
+      let text = this.$('#troll').text();
+      assert.equal(text, "Megatroll", 'the homepage template was rendered');
+    });
+  }
+
+  [`@test an alternate template will pull
+    in an alternate controller`](assert) {
```
Comment (mixonic) [inline packages/ember/tests/routing/decoupled_basic_test.js]: and here.

File: packages/ember/tests/routing/decoupled_basic_test.js [hunk 8]
```diff
@@ -0,0 +1,214 @@
...
+    }).then(() => {
+      assert.equal(this.currentPath, 'home');
+
+      let text = this.$('#app').text();
+      assert.equal(text, "Hours", 'the home template was rendered');
+    });
+  }
+
+  [`@test The Homepage with explicity 
+    template name in renderTemplate`](assert) {
```
Comment (mixonic) [inline packages/ember/tests/routing/decoupled_basic_test.js]: and another

File: packages/ember/tests/routing/decoupled_basic_test.js [hunk 9]
```diff
@@ -0,0 +1,214 @@
...
+    return this.visit('/').then(() => {
+      assert.equal(this.currentPath, 'home', 'currently on the home route');
+
+      let text = this.$('#app').text();
+      assert.equal(text, "Hours", 'the home template was rendered');
+    });
+  }
+
+  [`@test The Homepage and the Camelot 
+    page with multiple Router.map calls`](assert) {
```
Comment (mixonic) [inline packages/ember/tests/routing/decoupled_basic_test.js]: and here.

**Review by mixonic — CHANGES_REQUESTED** [review]
Small formatting things mostly. This is excellent!

File: packages/ember/tests/routing/decoupled_basic_test.js [hunk 10]
```diff
@@ -0,0 +1,329 @@
...
+    return this.visit('/').then(() => {
+      let text = this.$('p').text();
+
+      assert.equal(text, 'THIS IS THE REAL HOME',
+        'the homepage template was rendered'
+      );
+    });
+  }
+
+  ['@test defining templateName allows other templates to be rendered'](assert) {
```
Comment (patricksrobertson) [inline packages/ember/tests/routing/decoupled_basic_test.js]: From here on are additional tests.

Comment (patricksrobertson) [inline packages/ember/tests/routing/decoupled_basic_test.js]: @mixonic sorry, was currently adding more stuff to the file while you started the PR review. There are 5 more tests added at the end. Should be non-controversial edits.

File: packages/ember/tests/routing/decoupled_basic_test.js [hunk 11]
```diff
@@ -0,0 +1,329 @@
...
+      return this.visit('/');
+    }).then(() => {
+      assert.equal(this.currentPath, 'home');
+
+      let text = this.$('#app').text();
+      assert.equal(text, "Hours", 'the home template was rendered');
+    });
+  }
+
+  [`@test The Homepage with explicity template name in renderTemplate`](assert) {
```
Comment (mixonic) [inline packages/ember/tests/routing/decoupled_basic_test.js]: `explicity` => `explicit`

File: packages/ember/tests/routing/decoupled_basic_test.js [hunk 12]
```diff
@@ -0,0 +1,329 @@
...
+    return this.visit('/').then(() => {
+      let text = this.$('p').text();
+
+      assert.equal(text, 'Comes from homepage',
+        'the homepage template was rendered'
+      );
+    });
+  }
+
+  [`@test An alternate template will pull in an alternate controller instead of controllername`](assert) {
```
Comment (mixonic) [inline packages/ember/tests/routing/decoupled_basic_test.js]: `controllerName`? Weird test title anyway, not sure why any variant of `controllername` needs to be in it.

**Review by mixonic — APPROVED** [review]
I left two small test title comments, the rest of this looks fantastic. 🙇 Thank you for your patience.

**Comment by mixonic** [comment]
I'm kicking sauce to get things passing. We should be sure to remove the `decoupled_basic_test` back to `basic_test` when we're done here.

**Pull request merged**